File: G_The_Morning_Star.c
#include <iostream>
#include <unordered_map>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    for (int j = 0; j < t; ++j) {
        int n;
        cin >> n;
        
        unordered_map<int, int> countV, countH, countP, countN;
        long long res = 0;

        for (int i = 0; i < n; ++i) {
            int x, y;
            cin >> x >> y;

            // Add matches over vertical line
            res += countV[x];
            countV[x]++;

            // Add matches over horizontal line
            res += countH[y];
            countH[y]++;

            // Add matches over positive slope
            res += countP[x - y];
            countP[x - y]++;

            // Add matches over negative slope
            res += countN[x + y];
            countN[x + y]++;
        }

        cout << res * 2 << '\n';
    }

    return 0;
}

File: G_The_Morning_Star.py
from functools import lru_cache
from itertools import permutations, combinations, product, accumulate, cycle, chain
from collections import defaultdict,deque, Counter
from bisect import bisect,bisect_left,bisect_right
from math import sqrt,log10,log2,gcd,inf,perm,comb,ceil,floor
from sys import stdin,stdout
input = lambda: stdin.buffer.readline().decode().strip()
from heapq import heapify,heappop,heappush,heapreplace
#import sys
#sys.setrecursionlimit(10000)
##### ***** Frequently Used Code ***** #######
# list(map(int,input().split()))
# int(input())

for j in range(int(input())):
    n = int(input())
    countV = Counter()
    countH = Counter()
    countP = Counter() # +ve slope
    countN = Counter() # -ve slope
    res = 0
    for _ in range(n):
        x,y = map(int,input().split())
        # Add matches over vertical line
        res+=countV[x]
        countV[x]+=1
        # Add matches over horizantal line
        res+=countH[y]
        countH[y]+=1
        # Add matches over +ve slope
        #x-y = k
        res+=countP[x-y]
        countP[x-y]+=1
        # Add matches over -ve slope
        res+=countN[x+y]
        countN[x+y]+=1
    print(res*2)
    